types/dom-chromium-ai/index.d.ts

```diff
@@ -0,0 +1,253 @@
+interface WindowOrWorkerGlobalScope {
+    readonly ai: AI;
+}
+
+interface AI {
+    readonly assistant: AIAssistantFactory;
+    readonly summarizer: AISummarizerFactory;
+    readonly writer: AIWriterFactory;
+    readonly rewriter: AIRewriterFactory;
+}
+
+interface AICreateMonitor {
+    ondownloadprogress: (event: DownloadProgressEvent) => void;
+
+    addEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: AddEventListenerOptions | boolean,
+    ): void;
+    removeEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: EventListenerOptions | boolean,
+    ): void;
+}
+
+interface DownloadProgressEvent extends Event {
+    readonly loaded: number;
+    readonly total: number;
+}
+
+type AICreateMonitorCallback = (monitor: AICreateMonitor) => void;
+
+type AICapabilityAvailability = "readily" | "after-download" | "no";
+
+interface AIReadableStream extends ReadableStream<string> {
+    [Symbol.asyncIterator](): AsyncIterableIterator<string>;
+}
+
+// Assistant
+// https://github.com/explainers-by-googlers/prompt-api/#full-api-surface-in-web-idl
+
+interface AIAssistantFactory {
+    create(options?: AIAssistantCreateOptions): Promise<AIAssistant>;
+    capabilities(): Promise<AIAssistantCapabilities>;
+}
+
+interface AIAssistantCreateOptions {
+    signal?: AbortSignal;
+    monitor?: AICreateMonitorCallback;
+
+    systemPrompt?: string;
+    initialPrompts?: AIAssistantPrompt[];
```

Comment:
There's some [constraints regarding usage](https://github.com/explainers-by-googlers/prompt-api/#:~:text=Using%20both%20systemPrompt%20and%20a%20%7B%20role%3A%20%22system%22%20%7D%20prompt%20in%20initialPrompts%2C%20or%20using%20multiple%20%7B%20role%3A%20%22system%22%20%7D%20prompts%2C%20or%20placing%20the%20%7B%20role%3A%20%22system%22%20%7D%20prompt%20anywhere%20besides%20at%20the%200th%20position%20in%20initialPrompts%2C%20will%20reject%20with%20a%20TypeError.). I don't know if this can be expressed in TypeScript.

Reply:
I originally tried it, but gave up after a while. I think it's possible though.

Reply:
Should be fixed now!